src/state/apis/portals/portalsApi.ts

```diff
@@ -0,0 +1,771 @@
+                      balance.network as SupportedPortalsNetwork,
+                    )
+                    if (!chainId) return balanceAcc
+
+                    const assetId = toAssetId({
+                      chainId,
+                      assetNamespace: getAssetNamespaceFromChainId(chainId as KnownChainIds),
+                      assetReference: balance.address,
+                    })
+
+                    const stakedAmountCryptoBaseUnit = balance.rawBalance
+                    const fiatAmount = bnOrZero(balance.balanceUSD).toString()
+                    const apy = bnOrZero(balance.metrics?.apy ?? balance.metrics?.baseApy ?? 0)
+                      .div(100)
+                      .toString()
+                    const tvl = bnOrZero(balance.liquidity).toString()
+                    const icon = balance.image ?? balance.images?.[0] ?? ''
+                    const name = balance.name
+
+                    // Use #pool tag to determine if this is an LP token
+                    const isLpToken = balance.metadata?.tags?.includes('#pool')
+                    const defiType = isLpToken ? DefiType.LiquidityPool : DefiType.Staking
+
+                    const opportunityId: StakingId = `${balance.address}#${balance.key}`
+
+                    if (!balanceAcc.metadataByProvider[appName]) {
+                      balanceAcc.metadataByProvider[appName] = {
+                        provider: appName,
+                        icon: appImage,
+                        color: '#000000',
+                      }
+                    }
+
+                    const tokens = balance.tokens ?? []
+                    const underlyingAssetIds = tokens.map(tokenAddress =>
+                      toAssetId({
+                        chainId,
+                        assetNamespace: getAssetNamespaceFromChainId(chainId as KnownChainIds),
+                        assetReference: tokenAddress as string,
+                      }),
+                    )
+
+                    const assetMarketData = selectMarketDataByAssetIdUserCurrency(state, assetId)
+                    if (assetMarketData.price === '0' && balance.price) {
+                      dispatch(
+                        marketDataSlice.actions.setCryptoMarketData({
+                          [assetId]: {
+                            price: bnOrZero(balance.price).toString(),
+                            marketCap: '0',
+                            volume: bnOrZero(balance.metrics?.volumeUsd1d).toString(),
+                            changePercent24Hr: 0,
+                          },
+                        }),
+                      )
+                    }
+
+                    // Create or update the asset if it doesn't exist
+                    if (!assets[assetId]) {
+                      const icons = [...(balance.images ?? []), balance.image].filter(
+                        Boolean,
+                      ) as string[]
```

Comment:
Suggestion: We can use our type-safe `isSome` helper here to avoid the need to cast.

```suggestion
                      const icons = [...(balance.images ?? []), balance.image].filter(isSome)
```

Reply:
https://github.com/shapeshift/web/pull/8699/commits/d5bd4572a8acefed303e0b480ab77eaad8135fa9